frontend/src/app.js

```diff
@@ -215,6 +224,10 @@ search.addWidgets([
   }),
 ]);
 
+search.on('error', ({ error }) => {
+  TempStorage.searchError = error;
+});
```

Comment:
Why not to just set that `innerHtml` from here? Introducing a temp storage seems to be superfluous.

Reply:
Or else, you can introduce your custom widget for the error message, similar to what was done for the loading indicator. That would be even better if possible. Here you will be able to apply all your good architectural skills, and at the same time would not unnecessarily touch the other widgets.

Reply:
1st comment reply: i am using that searchError temp variable in newly created custom widget ("appInfiniteHits"). 
there is a use case where error message is visible and now i need to hide it because next request is success and we got hits with no error.
so in this use case i am conditionally checking if we have error show error message else restore default UI. hope this make sense to you.

2nd comment reply: i have tried a lot to gather sufficient information to pass this test in a very limited time. but i have no prior knowledge and experience in algoria, so i wouldn't be able to help you on this because its more like research and hit and trial work.

Reply:
1. I understand your intention now. However it should be achieved by another means. The `infinteHits` widget is just a representation module which sits on top of the `search` object and presents the results to the user. And this search object has other means of catching the search state - which you need to hide the error message. For example, why not to use [the `render` event](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#widget-param-render) of the search object? I agree that the documentation is not thorough enough, it lacks description of the parameters of the event. But what I usually do is logging the arguments into console and exploring it. Give it a try.